test/views/OverseasPensionTransferTaxChargeSpec.scala

```diff
@@ -0,0 +1,198 @@
+          if (userScenario.isAgent) anAgentUser else aUser,
+          if (userScenario.isAgent) fakeAgentRequest else fakeIndividualRequest)
+        implicit val htmlFormat = underTest(form(agentOrIndividual), taxYearEOY)
+        implicit val document: Document = Jsoup.parse(htmlFormat.body)
+
+        titleCheck(userScenario.specificExpectedResults.get.expectedTitle, userScenario.isWelsh)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY), Selectors.captionSelector)
+        radioButtonCheck(userScenario.commonExpectedResults.yes, 1, checked = false)
+        radioButtonCheck(userScenario.commonExpectedResults.no, 2, checked = false)
+        buttonCheck(userScenario.commonExpectedResults.continue)
+      }
+      "render the page with pre filled data" which {
+
+        val agentOrIndividual = if (userScenario.isAgent) "agent" else "individual"
+
+        implicit val messages: Messages = getMessages(userScenario.isWelsh)
+
+        implicit val userSessionDataRequest: UserSessionDataRequest[AnyContent] = new UserSessionDataRequest(aPensionsUserData,
+          if (userScenario.isAgent) anAgentUser else aUser,
+          if (userScenario.isAgent) fakeAgentRequest else fakeIndividualRequest)
+        implicit val htmlFormat = underTest(form(agentOrIndividual).bind(Map(RadioButtonAmountForm.yesNo -> "true", RadioButtonAmountForm.amount2 -> "100.00")), taxYearEOY)
+        implicit val document: Document = Jsoup.parse(htmlFormat.body)
+
+
+        titleCheck(userScenario.specificExpectedResults.get.expectedTitle, userScenario.isWelsh)
+        captionCheck(userScenario.commonExpectedResults.expectedCaption(taxYearEOY), Selectors.captionSelector)
+        textOnPageCheck(userScenario.commonExpectedResults.expectedAmountText, Selectors.amountLabelSelector)
+        inputFieldValueCheck("amount-2", Selectors.amountTextSelector, "100")
+        radioButtonCheck(userScenario.commonExpectedResults.yes, 1, checked = true)
+        radioButtonCheck(userScenario.commonExpectedResults.no, 2, checked = false)
+        buttonCheck(userScenario.commonExpectedResults.continue)
+      }
+
+      "render the page with an error when the user doesn’t select a radio button" which {
+        val agentOrIndividual = if (userScenario.isAgent) "agent" else "individual"
+        implicit val messages: Messages = getMessages(userScenario.isWelsh)
+        implicit val userSessionDataRequest: UserSessionDataRequest[AnyContent] = new UserSessionDataRequest(aPensionsUserData, aUser, if (userScenario.isAgent) fakeAgentRequest else fakeIndividualRequest)
+        implicit val htmlFormat = underTest(form(agentOrIndividual).bind(Map(RadioButtonAmountForm.yesNo -> "")), taxYearEOY)
+        implicit val document: Document = Jsoup.parse(htmlFormat.body)
+
+        errorSummaryCheck(userScenario.specificExpectedResults.get.expectedNoEntryErrorText, "#value")
```

Comment:
You may also want to check "errorAboveElementCheck" for all of the cases where relevant.